### PR TITLE
[webview_flutter] Add new entrypoint that uses hybrid composition on Android

### PR DESCRIPTION
**Pull request by bparrishMines**
## Related Issues

Fixes https://github.com/flutter/flutter/issues/61133

Tests on stable will fail until hybrid views support moves to stable.

To try early, add the dependency to your `pubspec.yaml` and switch to Flutter's master branch.
```yaml
dependencies:
  webview_flutter:
    git:
      url: git@github.com:bparrishMines/plugins.git
      path: packages/webview_flutter
      ref: webview_entrypoint
```

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/webview_flutter/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,27 @@
+## 0.3.24
+
+* Add support for building `WebView` widget with Android hybrid views. To use this feature, set
```
Comment (blasten) [inline packages/webview_flutter/CHANGELOG.md:3]: Android hybrid views should have link to https://github.com/flutter/flutter/wiki/Hybrid-Composition

**Comment by blasten** [comment]
SDK bump: `flutter: ">=1.22.0"` and version is `0.4.0`. 

@amirh are there any concerns with bumping the Flutter SDK constraint in pubspec.yaml?

File: packages/webview_flutter/pubspec.yaml [hunk 2]
```diff
@@ -1,11 +1,11 @@
 name: webview_flutter
 description: A Flutter plugin that provides a WebView widget on Android and iOS.
-version: 0.3.24
+version: 1.0.0
 homepage: https://github.com/flutter/plugins/tree/master/packages/webview_flutter
 
 environment:
   sdk: ">=2.7.0 <3.0.0"
-  flutter: ">=1.12.13+hotfix.5"
+  flutter: ">=1.22.0"
```
Comment (blasten) [inline packages/webview_flutter/pubspec.yaml:8]: It looks like this is `>=1.22.0-10.0.pre <2.0.0` in Maps.

**Review by blasten — APPROVED** [review]
LGTM

**Review by amirh — APPROVED** [review]
LGTM

File: packages/webview_flutter/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,71 @@
...
+https://github.com/flutter/flutter/issues/37989,
+https://github.com/flutter/flutter/issues/27924.
+
+* ✅ Text selection: https://github.com/flutter/flutter/issues/24585, https://github.com/flutter/flutter/issues/24584.
+
+* 🌎 API level: `WebView` just requires API level 19 (It used to be 20): https://github.com/flutter/flutter/issues/23728.
+
+* ⚡️ Performance: https://github.com/flutter/flutter/issues/61280, https://github.com/flutter/flutter/issues/31243.
+
+To get these fixes, set `WebView.platform = SurfaceAndroidWebView();` in `initState()`. For example:
```
[the remark this answers is not in the document]
Comment (bparrishMines) [reply packages/webview_flutter/CHANGELOG.md]: I think this was an old commit. It already says it's outdated

**Pull request merged**